File: DSA_Level-1/Recursion and Backtracking/Recursion On The Way Up/Print Kpc/PrintKpc_Solution.js

let keypad=[".;","abc","def","ghi","jkl","mno","pqrs","tu","vwx","yz"]; 

function printKpc(str,ans){
    if(str.length==0){
        console.log(ans);
        return; 
    }
    
    let code=str[0];
    let subStr=str.substring(1);
    
    let strCode=keypad[code-'0'];
    for(let i=0;i<strCode.length;++i){
        printKpc(subStr,ans+strCode[i]);
    }
}

printKpc("78","")

File: DSA_Level-1/Recursion and Backtracking/Recursion On The Way Up/Print Kpc/PrintKpc_Solution.java

import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner scn=new Scanner(System.in);
        String str=scn.nextLine();
        printKPC(str,"");
    }

    static String[] keypad={".;","abc","def","ghi","jkl","mno","pqrs","tu","vwx","yz"}; 
    public static void printKPC(String str, String asf) {
        
        if(str.length()==0){
            System.out.println(asf);
            return; 
        }
        
        char code=str.charAt(0);
        String subStr=str.substring(1);
        
        String strCode=keypad[code-'0'];
        for(int i=0;i<strCode.length();++i){
            printKPC(subStr,asf+strCode.charAt(i));
        }
    }

}